csm_web/frontend/src/components/course/ManageLabelsModal.tsx

```diff
@@ -0,0 +1,207 @@
+      setCourseLabels(data.sort((a: Label, b: Label) => a.id - b.id));
+      setCurrCourseLabels(data.sort((a: Label, b: Label) => a.id - b.id));
+    });
+  }, []);
+
+  const removeLabel = (id: number) => {
+    const tempLabels = courseLabels.filter((label: Label) => label.id !== id);
+    setCourseLabels(tempLabels);
+  };
+
+  const editRow = (id: number, name: string, description: string, showPopup: boolean) => {
+    const tempLabels = courseLabels.filter((label: Label) => label.id !== id);
+    const oldLabel = courseLabels.find((label: Label) => label.id === id);
+    const newLabel = { ...oldLabel, name: name, description: description, showPopup: showPopup } as Label;
+    let newLabels = [...tempLabels, newLabel];
+    const positiveLabelsSorted = newLabels.filter((label: Label) => label.id >= 0).sort((a, b) => a.id - b.id);
+    const negativeLabelsSorted = newLabels.filter((label: Label) => label.id < 0).sort((a, b) => b.id - a.id);
+    newLabels = [...positiveLabelsSorted, ...negativeLabelsSorted];
+    setCourseLabels(newLabels);
+  };
+
+  const handleNewRow = () => {
+    const course = courseLabels[0].course;
+    setCourseLabels([
+      ...courseLabels,
+      { id: newLabelKey, course: course, sections: [], name: "", description: "", showPopup: false }
+    ]);
+    setNewLabelKey(newLabelKey - 1);
+  };
+
+  function handleSubmit(event: React.ChangeEvent<HTMLFormElement>) {
+    event.preventDefault();
+    const currIDs = courseLabels.map((label: Label) => label.id);
+    const prevIDs = currCourseLabels.map((label: Label) => label.id);
+    const deletedIDs = [];
+    for (const id of prevIDs) {
+      if (!currIDs.includes(id)) {
+        deletedIDs.push(id);
+      }
+    }
+    const put_value = {
+      labels: courseLabels,
+      deletedLabelIds: deletedIDs
+    };
+    fetchWithMethod(`/courses/${courseId}/labels`, HTTP_METHODS.PUT, put_value).then(() => {
+      closeModal();
+      reloadSections();
+    });
+  }
+
+  return (
+    <Modal closeModal={closeModal}>
+      <form className="manage-labels-form" onSubmit={handleSubmit}>
+        <div>
+          <div className="manage-labels-form-header">Labels for {title}</div>
+          <div className="manage-labels-modal">
+            <ul>
+              {courseLabels.map((label: Label) => (
+                <EditLabelRow
+                  key={label.id}
+                  label={label}
+                  onChangeRow={(name: string, description: string, showPopup: boolean) =>
+                    editRow(label.id, name, description, showPopup)
+                  }
+                  removeLabel={() => removeLabel(label.id)}
+                />
+              ))}
+            </ul>
+          </div>
+          <div className="manage-label-modal-buttons">
+            <input className="affinity-btn" type="button" onClick={handleNewRow} value="Add New Label" />
+            <input className="affinity-btn" type="submit" value="Save Labels" />
```

Comment:
These buttons seem to be different from other buttons elsewhere; if changes are needed to existing buttons for alignment (which I believe was the cause for the custom class?), the appearance should still be as close as possible to the original.